hack/tests/resources/kubemark-machine-deployment-external-cluster.yaml

```diff
@@ -0,0 +1,66 @@
+apiVersion: cluster.x-k8s.io/v1beta1
+kind: MachineDeployment
+metadata:
+  annotations:
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-max-size: "5"
+    cluster.x-k8s.io/cluster-api-autoscaler-node-group-min-size: "1"
+  name: km-wl-kubemark-external-md-0
+  namespace: default
+spec:
+  clusterName: km-cp
+  replicas: 4
+  selector:
+    matchLabels:
+      cluster.x-k8s.io/cluster-name: km-cp
+      cluster.x-k8s.io/deployment-name: km-wl-kubemark-external-md-0
+  template:
+    metadata:
+      labels:
+        cluster.x-k8s.io/cluster-name: km-cp
+        cluster.x-k8s.io/deployment-name: km-wl-kubemark-external-md-0
+    spec:
+      bootstrap:
+        configRef:
+          apiVersion: bootstrap.cluster.x-k8s.io/v1beta1
+          kind: KubeadmConfigTemplate
+          name: km-wl-kubemark-external-md-0
+      clusterName: km-cp
+      infrastructureRef:
+        apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+        kind: KubemarkMachineTemplate
+        name: km-wl-kubemark-external-md-0
+      version: 1.21.1
+---
+apiVersion: infrastructure.cluster.x-k8s.io/v1alpha4
+kind: KubemarkMachineTemplate
+metadata:
+  labels:
+    cluster.x-k8s.io/cluster-name: km-cp
+  name: km-wl-kubemark-external-md-0
+  namespace: default
+spec:
+  template:
+    spec:
+      extraMounts:
+        - name: containerd-sock
+          containerPath: /run/containerd/containerd.sock
+          hostPath: /run/containerd/containerd.sock
+          type: Socket
+      kubemarkClusterSecretRef:
+        apiVersion: v1
+        kind: Secret
+        name: kubemark-cluster-kubeconfig
+        namespace: capk-system
```

Comment:
this is how we provide a reference to a secret for any external cluster intended for running kubemark pods